File: backend-C#/DoctorsOffice/Repositories/VisitRepository.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using DoctorsOffice.DbModels;
using Microsoft.EntityFrameworkCore;

namespace DoctorsOffice.Repositories
{
    public class VisitRepository
    {
        private readonly MyDbContext _context;

        public VisitRepository(MyDbContext context)
        {
            _context = context;
        }

        public ICollection<Visit> FindAllByDoctorPwzAndDates(string pwz, DateTime dateFrom, DateTime dateTo)
        {
            return _context.Visits
                .Where(visit =>
                    visit.Doctor.Pwz == pwz && visit.DateTime > dateFrom && visit.DateTime < dateTo)
                .ToList();
        }

        public Visit Save(Visit visit)
        {
            var addedVisit = _context.Visits.Add(visit).Entity;
            _context.SaveChanges();
            return addedVisit;
        }

        public List<Visit> FindForPatient(string identityNumber, string firstName, string lastName)
        {
            return _context.Visits
                .Include(v => v.Doctor)
                .Include(v => v.Facility)
                .Include(v => v.Patient)
                .Where(visit =>
                    visit.Patient.IdentityNumber == identityNumber &&
                    visit.Patient.FirstName == firstName &&
                    visit.Patient.LastName == lastName)
                .ToList();
        }
    }
}

File: backend-Java/src/main/java/pl/kacste/visit/VisitRepository.java
package pl.damkac.visit;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.time.LocalDateTime;
import java.util.List;

interface VisitRepository extends JpaRepository<Visit, Long> {

    @Query("select v from Visit v where v.doctor.pwz = :pwz and v.dateTime > :dateFrom and v.dateTime < :dateTo")
    List<Visit> finaAllByDoctorPwzAndDates(String pwz, LocalDateTime dateFrom, LocalDateTime dateTo);

    @Query("SELECT v from Visit v where v.patient.identityNumber = :identityNumber and v.patient.firstName = :firstName and v.patient.lastName = :lastName")
    List<Visit> findForPatient(String identityNumber, String firstName, String lastName);
}
